api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClient.java

```diff
@@ -52,18 +55,39 @@ public Mono<InventoryResponseDTO> addInventory(final InventoryRequestDTO model){
 
 
     public Mono<InventoryResponseDTO> updateInventory(InventoryRequestDTO model, String inventoryId){
-        return webClientBuilder.build().put()
+        return webClient.put()
                 .uri(inventoryServiceUrl + "/{inventoryId}" , inventoryId)
                 .body(Mono.just(model),InventoryRequestDTO.class)
                 .accept(MediaType.APPLICATION_JSON)
                 .retrieve().bodyToMono(InventoryResponseDTO.class);
     }
     public Mono<Void> deleteProductInInventory(final String inventoryId, final String productId){
-        return webClientBuilder.build().delete()
+        return webClient.delete()
                 .uri(inventoryServiceUrl + "/{inventoryId}/products/{productId}", inventoryId, productId)
                 .retrieve()
                 .bodyToMono(Void.class);
     }
+
+    public Flux<ProductResponseDTO> getProductsInInventoryByInventoryIdAndProductsField(final String inventoryId, final String inventoryName, final Double productPrice, final Integer productQuantity){
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUriString(inventoryServiceUrl + "/{inventoryType}/products")
+                .queryParamIfPresent("productName", Optional.ofNullable(inventoryName))
+                .queryParamIfPresent("productPrice", Optional.ofNullable(productPrice))
+                .queryParamIfPresent("productQuantity", Optional.ofNullable(productQuantity));
+
+        return webClient.get()
+                .uri(uriBuilder.buildAndExpand(inventoryId).toUri())
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
+                .bodyToFlux(ProductResponseDTO.class);
+    }
+
+    public Flux<InventoryResponseDTO> getAllInventory(){
+        return webClient.get()
+                .uri(inventoryServiceUrl)
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
```

Comment:
You should maybe specify the return, and check the number of items in the list to ensure correct and profound testing. Especially with the response EventTextStream to make sure its returning a FLUX